module/proto/gravity/v1/msgs.proto

```diff
@@ -1,237 +1,176 @@
+  string eth_signer           = 2;
+  bytes  signature            = 3 [
+    (gogoproto.casttype) = "github.com/ethereum/go-ethereum/common/hexutil.Bytes"
   ];
-  string ethereum_sender = 5;
-  string cosmos_receiver = 6;
-  string orchestrator    = 7;
 }
 
-message MsgDepositClaimResponse {}
+message MsgSubmitEthereumSignatureResponse {}
+
+// MsgSubmitEthereumEvent
+message MsgSubmitEthereumEvent {
+  option (gogoproto.goproto_getters) = false;
 
-// WithdrawClaim claims that a batch of withdrawal
-// operations on the bridge contract was executed.
-message MsgWithdrawClaim {
-  uint64 event_nonce    = 1;
-  uint64 block_height   = 2;
-  uint64 batch_nonce    = 3;
-  string token_contract = 4;
-  string orchestrator   = 5;
+  google.protobuf.Any event  = 1 [(cosmos_proto.accepts_interface) = "EthereumEvent"];
+  string              signer = 2;
 }
 
-message MsgWithdrawClaimResponse {}
-
-// ERC20DeployedClaim allows the Cosmos module
-// to learn about an ERC20 that someone deployed
-// to represent a Cosmos asset
-message MsgERC20DeployedClaim {
-  uint64 event_nonce    = 1;
-  uint64 block_height   = 2;
-  string cosmos_denom   = 3;
-  string token_contract = 4;
-  string name           = 5;
-  string symbol         = 6;
-  uint64 decimals       = 7;
-  string orchestrator   = 8;
+// SendToCosmosEvent is submitted when the SendToCosmosEvent is emitted by they gravity contract. 
+// ERC20 representation coins are minted to the cosmosreceiver address.
+message SendToCosmosEvent {
+  uint64 event_nonce     = 1;
+  string token_contract  = 2;
+  string amount          = 3 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int", 
+    (gogoproto.nullable) = false
+  ];
+  string ethereum_sender = 4;
+  string cosmos_receiver = 5;
+  uint64 ethereum_height = 6;
+}
+
+// BatchTxExecutedEvent claims that a batch of BatchTxExecutedal operations on the bridge contract
+// was executed successfully on ETH
+message BatchTxExecutedEvent {
+  string token_contract  = 1;
+  uint64 event_nonce     = 2;
+  uint64 ethereum_height = 3;
 }
 
-message MsgERC20DeployedClaimResponse {}
+// ContractCallTxExecutedEvent describes a contract call that has been
+// successfully executed on Ethereum.
 
-// This informs the Cosmos module that a logic
-// call has been executed
-message MsgLogicCallExecutedClaim {
+// NOTE: bytes.HexBytes is supposed to "help" with json encoding/decoding investigate?
+message ContractCallTxExecutedEvent {
   uint64 event_nonce        = 1;
-  uint64 block_height       = 2;
-  bytes  invalidation_id    = 3;
-  uint64 invalidation_nonce = 4;
-  string orchestrator       = 5;
+  bytes  invalidation_id    = 2 [
+    (gogoproto.casttype) = "github.com/tendermint/tendermint/libs/bytes.HexBytes"
+  ];
+  uint64 invalidation_nonce = 3;
+  uint64 ethereum_height    = 4;
+}
+
+// CosmosERC20DeployedEvent is submitted when an ERC20 contract
+// for a Cosmos SDK coin has been deployed on Ethereum.
+message CosmosERC20DeployedEvent {
```

Comment:
Not sure we need to call this "CosmosERC20DeployedEvent" instead of just "ERC20DeployedEvent". There is no need to disambiguate different ERC20Deployed events